firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/InAppMessageStreamManager.java

```diff
@@ -232,22 +244,38 @@ private boolean shouldIgnoreCache(String event) {
                       .defaultIfEmpty(CampaignImpressionList.getDefaultInstance())
                       .onErrorResumeNext(Maybe.just(CampaignImpressionList.getDefaultInstance()));
 
-              Function<CampaignImpressionList, Maybe<FetchEligibleCampaignsResponse>> serviceFetch =
-                  impressions ->
-                      taskToMaybe(apiClient.getFiams(impressions))
-                          .doOnSuccess(
-                              resp ->
-                                  Logging.logi(
-                                      String.format(
-                                          Locale.US,
-                                          "Successfully fetched %d messages from backend",
-                                          resp.getMessagesList().size())))
-                          .doOnSuccess(
-                              resp -> impressionStorageClient.clearImpressions(resp).subscribe())
-                          .doOnSuccess(analyticsEventsManager::updateContextualTriggers)
-                          .doOnSuccess(testDeviceHelper::processCampaignFetch)
-                          .doOnError(e -> Logging.logw("Service fetch error: " + e.getMessage()))
-                          .onErrorResumeNext(Maybe.empty()); // Absorb service failures
+              Maybe<InstanceIdResult> getIID = Maybe.fromCallable(() -> {
+                Task<InstanceIdResult> t = firebaseInstanceId.getInstanceId();
+                // This should be safe since these callbacks never happen on the UI thread.
+                Tasks.await(t);
+                if(t.isSuccessful() && t.getResult() != null) {
+                  return t.getResult();
+                } else {
+                  return null;
```

Comment:
Previously you were returning an expired response here which would have cached an expired campaign. Now nothing would be cached. I assume that is acceptable too?
Worth checking what happens on IID fetch failure (say when there is no network) and whether we are able to eventually recover.